include/Tableau.hpp

```diff
@@ -0,0 +1,120 @@
+/*
+* This file is part of the MQT QMAP library which is released under the MIT license.
+* See file README.md or go to https://iic.jku.at/eda/research/ibm_qx_mapping/ for more information.
+*/
+
+#ifndef QMAP_TABLEAU_HPP
+#define QMAP_TABLEAU_HPP
+
+#include "QuantumComputation.hpp"
+#include "utils/logging.hpp"
+
+#include <cstdint>
+#include <fstream>
+#include <limits>
+#include <ostream>
+#include <utility>
+#include <vector>
+
+class Tableau {
+    using EntryType   = std::int32_t;
```

Comment:
Although this is not the biggest concern right now, but using 32bit for a single entry in a table that is supposed to be binary seems to be very wasteful. Are you actually taking advantage of these 31 bits in any way? If not, then please consider choosing a more suitable type here. Technically, the most performant solution would probably be to use individual bits for every entry. But even `std::uint8_t` already reduces the memory footprint by a factor of 4.